airflow/runtimes/host_interrogator.go

```diff
@@ -0,0 +1,30 @@
+package runtimes
+
+type HostInterrogator interface {
+	IsMac() bool
+	IsWindows() bool
+	FileExists(string) bool
+	GetEnvVar(string) string
+}
+
+type hostInterrogator struct {
+	OSChecker
+	FileChecker
+	EnvChecker
+}
+
+func CreateHostInspector(osChecker OSChecker, fileChecker FileChecker, envChecker EnvChecker) HostInterrogator {
+	return hostInterrogator{
+		osChecker,
+		fileChecker,
+		envChecker,
+	}
+}
+
+func CreateHostInspectorWithDefaults() HostInterrogator {
+	return hostInterrogator{
+		CreateOSChecker(),
+		CreateFileChecker(),
+		CreateEnvChecker(),
+	}
+}
```

Comment:
Instead of passing all 3 of these implementations around, I've consolidated them into a `HostInterrogator` object that can answer all the questions about the host. In our tests we mock out anything that interacts with the underlying host.